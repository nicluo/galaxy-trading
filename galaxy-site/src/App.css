.App {
}

.App-header {
  background-color: #222;
  height: 3rem;
  padding: 0.5rem;
  color: white;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 2rem;
  display: inline-block;
  margin-right: 1rem;
  vertical-align: middle;
}

.App-title {
  font-size: 1.5rem;
  display: inline-block;
  vertical-align: middle;
  margin: 0;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}
