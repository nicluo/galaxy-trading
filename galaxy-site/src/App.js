import React, { Component } from 'react';
import ResultList from './ResultList';
import Form from './Form';
import logo from './logo.svg';
import 'bootstrap/dist/css/bootstrap.css';
import './App.css';
import {createSession, getSession} from './lib/api';
import './lib/localstorage';
import {getSessionId, setSessionId} from "./lib/localstorage";

class App extends Component {
  state = {
    session: null
  };

  initializeSession() {
    if(getSessionId() !== null) {
      getSession(getSessionId())
        .then(session => this.setState({ session }))
        .catch(() => this.startNewSession());
    } else {
      this.startNewSession();
    }
  }

  startNewSession() {
    createSession()
      .then(session => {
        this.setState({ session });
        setSessionId(session.id);
      });
  }

  componentDidMount() {
    this.initializeSession();
  }

  render() {
    const {session} = this.state;

    return (
      <div className="App">
        <header className="App-header">
          <div className="text-white float-right">{ session ? session.id : '' }</div>
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Space Math</h1>
        </header>

        <ResultList session={session} />
        <Form session={session} />
      </div>
    );
  }
}

export default App;
