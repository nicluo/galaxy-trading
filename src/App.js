import React, { Component } from 'react';
import logo from './logo.svg';
import 'bootstrap/dist/css/bootstrap.css';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Space Math</h1>
        </header>

        {/* ResultList Component */}
        <div className="container my-4">
          <div className="card my-2">
            <ul className="list-group list-group-flush">
              <li className="list-group-item">
                <strong className="h4">
                  how much wood could a woodchuck chuck if a woodchuck could chuck wood ?
                </strong>
              </li>
              <li className="list-group-item list-group-item-danger">
                I have no idea what you are talking about
              </li>
            </ul>
          </div>
          <div className="card my-2">
            <ul className="list-group list-group-flush">
              <li className="list-group-item">
                <strong className="h4">
                  how much is pish tegj glob glob ?
                </strong>
              </li>
              <li className="list-group-item list-group-item-success">
                pish tegj glob glob is 42
              </li>
            </ul>
          </div>
        </div>

        {/* Form Component */}
        <div className="container my-4">
          <div className="form-group">
            <input type="text" className="form-control form-control-lg" id="query" aria-describedby="query" placeholder="how much is pish pish Iron ?" autoFocus="" />
          </div>
        </div>
      </div>
    );
  }
}

export default App;
