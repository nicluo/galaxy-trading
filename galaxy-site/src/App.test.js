import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

it('renders <App /> loading state', () => {
  const wrapper = shallow(<App sessionId={null} offline={false} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it('renders <App /> offline state', () => {
  const wrapper = shallow(<App sessionId={null} offline={true} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});