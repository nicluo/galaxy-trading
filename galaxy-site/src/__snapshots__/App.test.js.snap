// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <App /> loading state 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="text-white float-right"
    />
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Space Math
    </h1>
  </header>
  <div
    className="alert alert-info"
    role="alert"
  >
    Connecting to API server...
  </div>
</div>
`;

exports[`renders <App /> offline state 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="text-white float-right"
    />
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Space Math
    </h1>
  </header>
  <div
    className="alert alert-danger"
    role="alert"
  >
    Unable to connect to API server.
  </div>
</div>
`;
